src/test/java/com/skynest/EditBucketTest.java

```diff
@@ -6,27 +6,73 @@
 import org.testng.annotations.Test;
 
-import static org.apache.http.HttpStatus.SC_OK;
+import java.util.UUID;
+
+import static org.apache.http.HttpStatus.*;
 import static org.testng.Assert.assertEquals;
 
 public class EditBucketTest extends BucketBaseTest {
 
-    @Test(dataProvider = "BucketData")
-    void logged_admin_should_be_able_to_edit_specific_bucket(EditBucketRequest editBucketRequest) {
-        Response response = skyNestBackendClient.editBucket(editBucketRequest, createdBucketId);
+    @Test(dataProvider = "validBucketData")
+    void logged_owner_should_be_able_to_edit_private_bucket_with_valid_data(EditBucketRequest editBucketRequest) {
+        Response response = skyNestBackendClient.editBucket(editBucketRequest, createdBucketResponse.getBucketId());
+        response.then().statusCode(SC_OK);
+
+        BucketResponse editBucketResponse = response.as(BucketResponse.class);
+
+        assertEquals(editBucketResponse.getName(), editBucketRequest.getName());
+        assertEquals(editBucketResponse.getDescription(), editBucketRequest.getDescription());
+        assertEquals(editBucketResponse.getIsPublic(), editBucketRequest.getIsPublic());
+
+        createdBucketResponse = editBucketResponse;
+    }
+
+    @Test
+    void every_logged_user_should_be_able_to_edit_public_bucket_with_valid_data() {
+        loginAs(Roles.WORKER);
+        EditBucketRequest editBucketRequest = EditBucketRequest.generateEditRequestForPublicBucket();
+        Response response = skyNestBackendClient.editBucket(editBucketRequest, createdBucketResponse.getBucketId());
         response.then().statusCode(SC_OK);
 
         BucketResponse editBucketResponse = response.as(BucketResponse.class);
 
         assertEquals(editBucketResponse.getName(), editBucketRequest.getName());
         assertEquals(editBucketResponse.getDescription(), editBucketRequest.getDescription());
         assertEquals(editBucketResponse.getIsPublic(), editBucketRequest.getIsPublic());
+
+        createdBucketResponse = editBucketResponse;
+    }
+
+    @Test
+    void logged_user_should_not_be_able_to_edit_bucket_by_not_existing_id() {
+        EditBucketRequest editBucketRequest = EditBucketRequest.generateValidBucketEditRequest();
+        Response response = skyNestBackendClient.editBucket(editBucketRequest, UUID.randomUUID());
+        response.then().statusCode(SC_NOT_FOUND);
+    }
+
+    @Test(dataProvider = "invalidBucketData")
+    void logged_user_should_not_be_able_to_edit_specific_bucket_with_invalid_data(EditBucketRequest editBucketRequest) {
+        Response response = skyNestBackendClient.editBucket(editBucketRequest, createdBucketResponse.getBucketId());
+        response.then().statusCode(SC_BAD_REQUEST);
+    }
+
+    @DataProvider(name = "invalidBucketData")
+    public Object[][] getInvalidBucketData() {
+        return new Object[][]{
+                new Object[]{EditBucketRequest.generateValidBucketEditRequest().withName("")},
```

Comment:
same as before; consider having a variable for the duplicated string value and use that instead